src/ZkFold/Base/Algebra/Polynomials/Multivariate/Polynomial.hs

```diff
@@ -57,7 +63,9 @@ instance (FromPolynomial c i j m p, FromMonomial i j m) => Eq (P c i j m p) wher
 
 -- TODO: this assumes sorted monomials! Needs fixing.
 instance (FromPolynomial c i j m p, FromMonomial i j m) => Ord (P c i j m p) where
-    compare (P l) (P r) = compare (map snd $ fromPolynomial @c @i @j @m l) (map snd $ fromPolynomial @c @i @j @m r)
+    compare (P l) (P r) = compare
```

Comment:
We can make this even more readable by using `Data.Function.on`

Reply:
No it's more readable since we don't use `fromPolynomial` anymore.